yb-voyager/src/queryissue/unsupported_dml_constructs.go

```diff
@@ -111,3 +111,93 @@ func (d *XmlExprDetector) Detect(msg protoreflect.Message) ([]string, error) {
 	}
 	return nil, nil
 }
+
+type RangeTableFunDetector struct{}
+
+func NewRangeTableFuncDetector() *RangeTableFunDetector {
+	return &RangeTableFunDetector{}
+}
+
+// Detect checks if a RangeTableFunc node is present for a XMLTABLE() function
+func (d *RangeTableFunDetector) Detect(msg protoreflect.Message) ([]string, error) {
+	if queryparser.GetMsgFullName(msg) == queryparser.PG_QUERY_RANGETABLEFUNC_NODE {
+		if isXMLTable(msg) {
+			return []string{XML_FUNCTIONS}, nil
```

Comment:
Discussion for later: 
I;m wondering if it might be a good idea for `Detect` to not return anything, just store the detection information in it's struct; We declare another func for all detectors `GetIssues() []IssueInstance` that will generate the issues directly and give it. `Detector.GetIssues()` will be called at the end after the full proto message has been traverserd.   

Reason being: What we do now is combine is just get the constructTypes from each Detect() call, store it in one list, and then create `IssueInstances` out of it. This flow can't work when we want to add more details/metadata (like the function names, etc) to each specific issue.

Thoughts?

Reply:
yeah good idea, we can write getter functions of the Detectors.